### DISCUSSION TITLE
Supporting OAuth without exposing the PB server to the world?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [pocketbase/pocketbase#4203] [Q&A] [answered]
From what I understand, the user agent (browser) HAS to be able to redirect back to pbserver.com/api/oauth2-redirect right?

So that means my pb server has to be exposes to the internet to handle the redirect?

**Comment by ganigeorgiev** [comment, marked as answer]
You can define your own redirect handler if you want to but you'll need to use the ["Manual code exchange" flow](https://pocketbase.io/docs/authentication/#oauth2-integration). The example in the documentation shows how it can be done with a static client-side JS, but you can apply the same idea for any other implementation.

> my pb server has to be exposes to the internet to handle the redirect

I don't understand what is your concern with that but note that it is fine PocketBase to be directly accessible.
It was designed with the idea that you'll interact with the PocketBase web APIs mostly from client-side (ex. SPA in a browser, at least that is my use case).
You'll also need to expose it anyway if you plan to use the realtime API from the client-side, unless you decide to proxy the connection through another backend but this comes with a lot of caveats and I wouldn't recommend it unless you have a really good reason for it.

**Reply by VictorioBerra** [reply]
@ganigeorgiev thank you that helps. I was not aware exposing the PB server was generally OK to do. I am quite paranoid about security. Many FOSS apps like these have big red disclaimers cautioning people against exposing the product directly.
